### PR TITLE
add nextPos in canal.RowsEvent

### PR DESCRIPTION
**Pull request by ideamean**
ref #247 
hi, I add the nextPos in canal.RowsEvent, would you please check if it is ok?

### CODE REVIEW

File: canal/dump.go [hunk 1]
```diff
@@ -70,6 +70,8 @@ func (h *dumpParseHandler) Data(db string, table string, values []string) error
 	}
 
 	events := newRowsEvent(tableInfo, InsertAction, [][]interface{}{vs})
+	nextPos := h.c.syncer.GetNextPosition()
+	events.NextPos = nextPos
```
Comment (siddontang) [inline canal/dump.go]: this NextPos is not the next pos of current event. You can use the event header Pos directly.

Comment (ideamean) [reply canal/dump.go]: where is the event header pos? like this?
```
nextPos := mysql.Position{h.name, h.pos}
events.NextPos = nextPos
```

Comment (ideamean) [reply canal/dump.go]: ``
nextPos := mysql.Position{h.name, uint32(h.pos)}
```

Comment (siddontang) [reply canal/dump.go]: This is for dump, and I think `syncer.GetNextPosition` can get the correct position here. You can only get the binlog sync pos after dump.

File: canal/sync.go [hunk 2]
```diff
@@ -187,6 +187,7 @@ func (c *Canal) handleRowsEvent(e *replication.BinlogEvent) error {
 		return errors.Errorf("%s not supported now", e.Header.EventType)
 	}
 	events := newRowsEvent(t, action, ev.Rows)
+	events.NextPos = nextPos
```
Comment (siddontang) [inline canal/sync.go]: why not passing nextPos as the param of newRowsEvent?

File: canal/sync.go [hunk 3]
```diff
@@ -164,7 +164,7 @@ func (c *Canal) runSyncBinlog() error {
 	return nil
 }
 
-func (c *Canal) handleRowsEvent(e *replication.BinlogEvent) error {
+func (c *Canal) handleRowsEvent(e *replication.BinlogEvent, nextPos mysql.Position) error {
```
Comment (siddontang) [inline canal/sync.go]: no need nextPos here, you can still get it from `e.Header.LogPos`

File: canal/sync.go (hunk 2 above)
Comment (ideamean) [reply canal/sync.go]: ok, I will follow your guide

File: canal/dump.go (hunk 1 above)
Comment (ideamean) [reply canal/dump.go]: I'm so sorry I can't understand your opinion, which is right? 

nextPos := h.c.syncer.GetNextPosition()
or 
nextPos := mysql.Position{h.name, uint32(h.pos)}

if I add a param in "newRowsEvent(t, action, ev.Rows)",  should modify here too.

Comment (siddontang) [reply canal/dump.go]: both are not right here, here in the dump produce, we can't know anything about the sync position. So you can use the position here.

**Comment by GregoryIan** [comment]
@ideamean I will close this pr. If you still need it, welcome to open and update it again

**Pull request closed without merging**